iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -396,8 +396,8 @@ def save(self, outPath, outBasenameSuffix="", outzipFilePrefix=""):
                 for f in self.files:
                     self.dts.info("viewer:info", "Saving in output zip %s" % f.filename)
                     fout = attrdict(write=lambda s: zout.writestr(_outPrefix + f.filename, s))
-                    writer = XHTMLSerializer()
-                    writer.serialize(f.xmlDocument, fout)
+                    writer = XHTMLSerializer(fout)
```

Comment:
I don't know Python well, but do we need some kind of output stream buffering, or does ZipFile do that for us?

Reply:
Yes, I think you're right - this is broken.  The above code is relying on `fout` being written to in a single call to `write()` , which is not the norm for a Python "file-like object".

I think this needs to be rewritten to use `BytesIO`.

Reply:
This is now done in 52b1586773ea386ac8764f026dfd4b33d1ee08d8